src/Strategy/Cocoapods/PodfileLock.hs

```diff
@@ -209,58 +214,66 @@ buildGraph lockFilePath lockFile@PodLock{lockExternalSources} = do
           --  * Parent has git source
           --
           -- We replace this dependency with parent's git sourced dependency.
-          let (parentSpec, candidateSubSpec) = splitOnceOn "/" dependencyName
+          let (parentSpec, childSpec) = splitOnceOn "/" dependencyName
 
-          if Data.Text.null candidateSubSpec
+          if Data.Text.null childSpec
             then pure d
             else do
               revisedDep <- case Map.lookup parentSpec lockExternalSources of
-                Just (ExternalPodSpec podSpecPath) -> fromMaybe d <$> (readPodSubSpecSourceAt podSpecPath parentSpec)
-                Just (ExternalPath podPath) -> fromMaybe d <$> (readPodSubSpecSourceAt (toPodSpecPath podPath parentSpec) candidateSubSpec)
+                Just (ExternalPodSpec podSpecPath) ->
+                  (fromMaybe d <$> (readPodSubSpecSourceAt podSpecPath childSpec)) <||> pure d
+                Just (ExternalPath podPath) ->
+                  (fromMaybe d <$> (readPodSubSpecSourceAt (toPodSpecPath podPath parentSpec) childSpec))
+                    <||> pure d
```

Comment:
It feels a bit odd to have `<||> pure d` on both of these, particularly when they already have `fromMaybe d`.  It looks like the difference between the two is just what path to use.  Maybe this could be refactored into a few smaller steps to create less reuse and make all the alternatives clearer.